test/kubernetes_unittest.cc

```diff
@@ -509,4 +515,186 @@ TEST_F(KubernetesTest, GetContainerMetadata) {
   });
   EXPECT_EQ(expected_metadata->ToString(), m.metadata().metadata->ToString());
 }
+TEST_F(KubernetesTest, GetPodAndContainerMetadata) {
+  Configuration config(std::stringstream(
+    "KubernetesClusterName: TestClusterName\n"
+    "KubernetesClusterLocation: TestClusterLocation\n"
+    "MetadataApiResourceTypePerarator: \".\"\n"
+    "MetadataIngestionRawContentVersion: TestVersion\n"
+    "InstanceZone: TestZone\n"
+    "InstanceId: TestID\n"
+  ));
+  Environment environment(config);
+  KubernetesReader reader(config, nullptr);  // Don't need HealthChecker.
+
+  json::value controller = json::object({
+    {"controller", json::boolean(true)},
+    {"apiVersion", json::string("1.2.3")},
+    {"kind", json::string("TestKind")},
+    {"name", json::string("TestName")},
+    {"uid", json::string("TestUID1")},
+    {"metadata", json::object({
+      {"name", json::string("InnerTestName")},
+      {"kind", json::string("InnerTestKind")},
+      {"uid", json::string("InnerTestUID1")},
+    })},
+  });
+  UpdateOwnersCache(&reader, "1.2.3/TestKind/TestUID1", controller);
+  json::value pod = json::object({
+    {"metadata", json::object({
+      {"name", json::string("TestPodName")},
+      {"namespace", json::string("TestNamespace")},
+      {"uid", json::string("TestPodUid")},
+      {"creationTimestamp", json::string("2018-03-03T01:23:45.678901234Z")},
+    })},
+    {"spec", json::object({
+      {"nodeName", json::string("TestSpecNodeName")},
+      {"containers", json::array({
+        json::object({{"name", json::string("TestContainerName0")}}),
+      })},
+    })},
+    {"status", json::object({
+      {"containerID", json::string("docker://TestContainerID")},
+      {"containerStatuses", json::array({
+        json::object({
+          {"name", json::string("TestContainerName0")},
+        }),
+      })},
+    })},
+  });
+
+  auto metadatas = GetPodAndContainerMetadata(
```

Comment:
why not `m` like in other places.
If you want plural, then `metadata_list` or something is better.

Reply:
changed to metadata_list

Reply:
Frankly, I prefer `m` for brevity. Any reason to not go with that? As @supriyagarg mentioned, this is a convention in other tests.